app/graphql/mutations/system/associate_profiles.rb

```diff
@@ -12,8 +12,9 @@ class AssociateProfiles < BaseMutation
 
       def resolve(args = {})
         host = find_hosts([args[:id]]).first
-        profiles = find_profiles(args[:profile_ids])
-        host.update(profiles: profiles)
+        external_profiles = host.profiles.where(external: true)
+        internal_profiles = find_profiles(args[:profile_ids])
+        host.update(profiles: internal_profiles + external_profiles)
```

Comment:
I'd agree using `internal_profiles.or(external_profiles)` here is a better solution, but:
`ArgumentError: Relation passed to #or must be structurally compatible. Incompatible values: [:joins]`